src/openrct2/core/FlagHolder.hpp

```diff
@@ -0,0 +1,51 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2024 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+template<typename THolderType, typename TEnumType> struct FlagHolder
+{
+    THolderType holder = 0;
```

Comment:
Do we want these hardcoded to `0`? Or should we do all of them like: `TholderType holder = {}`?

Reply:
I have a habit of using `= 0` for integers, but I would have no problem using `holder{}` here if that’s what others prefer.

Reply:
I think its better to use brace initialization for generic types however not gonna make this a blocker.